ksqldb-engine/src/main/java/io/confluent/ksql/engine/rewrite/QueryMask.java

```diff
@@ -0,0 +1,170 @@
+  private static final ImmutableSet<String> DISALLOWED_KEYS = ImmutableSet.of("connection.url",
+      "mode", "topic.prefix", "table.whitelist", "key");
+  private static final String MASKED_VALUE = "'[string]'";
+
+  public QueryMask(final KsqlRequest request) {
+    maskedQuery = getMaskedStatement(request.getKsql());
+    System.out.println("here");
+  }
+
+  public String getMaskedQuery() {
+    return maskedQuery;
+  }
+
+  private String getMaskedStatement(final String query) {
+    try {
+      final ParseTree tree = DefaultKsqlParser.getParseTree(query);
+      return new Visitor().visit(tree);
+    } catch (final Exception e) {
+      return maskConnectProperties(query);
+    }
+  }
+
+  private String maskConnectProperties(final String query) {
+    /*
+      Failed to parse statement. Mask disallowed keys by string replacing. This is a best effort
+      attempt. The assumption is that properties are in right format:
+      "(\"[^\"]+\")\\s*=\\s*('[^']*'|\"[^\"]*\")")
+     */
+    final StringBuffer sb = new StringBuffer();
+    final Pattern pattern = Pattern.compile("(\"[^\"]+\")\\s*=\\s*('[^']*'|\"[^\"]*\")");
+    final Matcher matcher = pattern.matcher(query);
+
+    while (matcher.find()) {
+      final String matchedString = matcher.group(0);
+      final String key = matcher.group(1).substring(1, matcher.group(1).length() - 1);
+
+      if (DISALLOWED_KEYS.contains(key)) {
+        final String replacement = "\"" + key + "\"=" + MASKED_VALUE;
+        matcher.appendReplacement(sb, replacement);
+      } else {
+        matcher.appendReplacement(sb, matchedString);
+      }
+    }
+
+    matcher.appendTail(sb);
+
+    return "";
+  }
+
+  private static final class Visitor extends SqlBaseBaseVisitor<String> {
+
+    private boolean masked;
+
+    @Override
+    public String visitStatements(final StatementsContext context) {
+      final List<String> statementList = new ArrayList<>();
+      for (final SingleStatementContext stmtContext : context.singleStatement()) {
+
+        masked = false;
+        final String statement = visitSingleStatement(stmtContext);
+
+        if (masked) {
+          statementList.add(statement);
+        } else {
+          // Use original statement if not masked
+          final int start = stmtContext.start.getStartIndex();
+          final int end = stmtContext.stop.getStopIndex();
+          final Interval interval = new Interval(start, end);
+          final String original = stmtContext.start.getInputStream().getText(interval);
+          statementList.add(original);
+        }
+      }
+      return StringUtils.join(statementList, "\n");
+    }
+
+    @Override
+    public String visitSingleStatement(final SingleStatementContext context) {
+      return String.format("%s;", visit(context.statement()));
+    }
+
+    @Override
+    public String visitCreateConnector(final CreateConnectorContext context) {
```

Comment:
This looks like it's doing work that the SqlFormatter should be responsible for. I had a quick look at the SqlFormatter just now and it doesn't look like it implements `visitCreateConnector()`, which is odd. If it makes sense to unify the two, I think that would be better so that code for formatting sql statements only needs to be evolved in one place if the sql syntax ever changes in the future.